mapusaurus/hmda/views.py

```diff
@@ -28,38 +28,50 @@ def loan_originations(request):
     query = HMDARecord.objects.filter(base_hmda_query())
     if action_taken_selected:
         query = query.filter(action_taken__in=action_taken_selected)
-    if institution_selected:
-        if lender_hierarchy == 'true':
-            hierarchy_list = institution_selected.get_lender_hierarchy(False, False)
-            if len(hierarchy_list) > 0:
-                query = query.filter(institution__in=hierarchy_list) 
+    #if lender param key is passed in
+    if institution_id:
+        #get institution_selected
+        institution_selected = Institution.objects.filter(pk=institution_id).first()
+        #if param value is valid
+        if institution_selected: 
+            if lender_hierarchy == 'true':
+                hierarchy_list = institution_selected.get_lender_hierarchy(False, False)
+                if len(hierarchy_list) > 0:
+                    query = query.filter(institution__in=hierarchy_list) 
+                else: 
+                    query = query.filter(institution=institution_selected)
+            elif peers == 'true':
+                peer_list = institution_selected.get_peer_list(metro_selected, True, False)
+                if len(peer_list) > 0:
+                    query = query.filter(institution__in=peer_list)
+                else:
+                    query = query.filter(institution=institution_selected)
             else: 
                 query = query.filter(institution=institution_selected)
-        elif peers == 'true':
-            peer_list = institution_selected.get_peer_list(metro_selected, True, False)
-            if len(peer_list) > 0:
-                query = query.filter(institution__in=peer_list)
-            else:
-                query = query.filter(institution=institution_selected)
-        else: 
-            query = query.filter(institution=institution_selected)
-    else:
-        return HttpResponseBadRequest("Missing lender")
-    if geos is None:
-        return HttpResponseBadRequest("Missing one lat/lon bounds or metro.")
-    query = query.filter(geo__in=geos)
+        else:
+            return None
+    #if valid censustracts for metro or bounds.
+    if census_tracts is None:
+        return None
```

Comment:
Having the response be a queryset or None is a little wonky (not that a queryset or an HttpResponseBadRequest is better), as it requires all callers to account for both cases. I wonder if this isn't a good use case for exceptions?